### PR TITLE
Add ERC: Content encoding in ERC-5219 mode Web3 URL

### PR DESCRIPTION
**Pull request by nand2**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-7592.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+eip: 7592
```
Comment (abcoathup) [inline ERCS/erc-7592.md:2]:
```suggestion
eip: 7618
```
Assigning next sequential EIP/ERC/RIP number.  7592 was already assigned to another ERC.
Numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

**Review by Elenore23 — APPROVED** [review]

File: ERCS/erc-7618.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Motivation
+
+As storage in blockchains is expensive, it is optimal to try to store and serve compressed assets. Standard HTTP use the `Accept-Encoding`/`Content-Encoding` mechanism, in which the client specify their supported compression algorithms, and the server returns the data compressed in one of them. It is not optimal to replicate this mechanism in the `web3://` protocol, due to blockchain storage and computation constraints. Moreover, it is not possible to directly serve content with a `Content-Encoding` header, because we cannot know if the HTTP client implements the compression algorithm.
+
+By specifying a list of supported compression algorithms, doing the decompression at the protocol side and serving the decompressed data to the client, we can safely store compressed data and serve it.
+
+## Specification
+
+In the [ERC-6944](./eip-6944.md) resolve mode, this standard indicates that if a ``Content-encoding`` HTTP header (in the `headers` `KeyValue` array of the `request()` method) is provided, then the protocol will attempt to decode the content before forwarding it to the `web3://` client.
```
Comment (SamWilsn) [inline ERCS/erc-7618.md:26]: Please be consistent with the case of `Content-Encoding`.

File: ERCS/erc-7618.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+This standard indicate the list of supported content encodings: `gzip`, `br` (brotli). If the advertized ``Content-encoding`` is not part of this list, an error indicating an unsupported content encoding will be sent to the client.
```
Comment (SamWilsn) [inline ERCS/erc-7618.md:28]: Could you word this more like a requirement?

```suggestion
Gateways supporting this protocol MUST support at minimum `gzip` and `br` (brotli). Gateways MAY support additional encodings. If a gateway does not support the advertised encoding, it MUST return an error to the client.
```

**Comment by nand2** [comment]
I have modified the contents to take into account Sam's point : if the compression of the returned data is advertised by the client in `accept-encoding`, then decompression by the protocol is not necessary.
It is only necessary when the client does not advertise decompression capabilities for the compression of the returned data.

**Comment by nand2** [comment]
And thanks for your review @SamWilsn !

File: ERCS/erc-7618.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+By specifying a list of supported compression algorithms, optionally doing the decompression at the protocol side and serving the data to the client, we can safely store compressed data and serve it.
+
+## Specification
+
+In the [ERC-6944](./eip-6944.md) resolve mode, this standard indicates that if a ``Content-Encoding`` HTTP header (in the returned `headers` `KeyValue` array of the `request()` method) is provided, and if it is not part of the supported algorithms provided by the client in the ``Accept-Encoding`` header, or the client did not provide an ``Accept-Encoding`` header, then the protocol MUST decode the content before forwarding it to the `web3://` client.
+
+The protocol MUST support the following content encodings: `gzip`, `br` (brotli). If the protocol is to decode the content, and if the advertized ``Content-encoding`` is not part of this list, an error indicating an unsupported content encoding MUST be sent to the client. Once decoded, the decompressed data is sent to the client. The ``Content-Encoding`` header MUST NOT be forwarded to the client.
```
Comment (SamWilsn) [inline ERCS/erc-7618.md:28]:
```suggestion
The protocol MUST support the following content encodings: `gzip`, `br` (brotli). If the protocol is to decode the content, and if the advertized ``Content-encoding`` is not part of this list, an error indicating an unsupported content encoding MUST be sent to the client. Once decoded, the decompressed data is sent to the client. The ``Content-Encoding`` header MUST NOT be forwarded to the client when the protocol decodes the content.
```

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**